main.go

```diff
@@ -287,7 +294,7 @@ func vectorHandler(defaultSRV string, probers []prober.Prober) func(http.Respons
 			http.Error(w, err.Error(), http.StatusServiceUnavailable)
 			return
 		}
-		lats := getLatencies(r.Context(), probers, urls)
+		lats := getLatencies(r.Context(), probers, urls, timeout)
```

Comment:
It would have to be done here i guess, create a descendent context that has the tonight built in